File: LabPoo/src/main/herencia.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Admin = exports.Estudiante = void 0;
const encapsulamiento_1 = require("./encapsulamiento");
class Estudiante extends encapsulamiento_1.Persona {
    constructor(nombre, edad, sexo, matricula) {
        super(nombre, edad, sexo);
        this.matricula = matricula;
    }
    getMatricula() {
        return this.matricula;
    }
    setMatricula(matricula) {
        this.matricula = matricula;
    }
}
exports.Estudiante = Estudiante;
class Admin extends encapsulamiento_1.Usuario {
    constructor(nombre, edad, sexo, permisos) {
        super(nombre, edad, sexo, 'Administrador');
        this.permisos = permisos;
    }
    describir() {
        return `Admin: ${this.nombre}, Edad: ${this.edad}, Permisos: ${this.permisos.join(', ')}`;
    }
    agregarPermiso(permiso) {
        this.permisos.push(permiso);
    }
    mostrarPermisos() {
        return `Permisos de ${this.nombre}: ${this.permisos.join(', ')}`;
    }
}
exports.Admin = Admin;

File: tests/herencia.py
from encapsulamiento import Persona

class Estudiante(Persona):    
    def __init__(self, nombre, edad, carrera):
        super().__init__(nombre, edad)
        self.carrera = carrera

    def saludar(self):
        print(f'Hola, soy {self.get_nombre()} y estudio {self.carrera}.')

    def get_carrera(self):
        return self.carrera

    def set_carrera(self, carrera):
        self.carrera = carrera
    
exports = [Estudiante]